main.go

```diff
@@ -3,81 +3,98 @@ package main
 import (
 	"context"
 	"flag"
+	"fmt"
 	"log"
-	"net"
 
 	"github.com/armon/go-socks5"
 )
 
 const (
-	allowed = true
-	denied  = false
+	allowed            = true
+	denied             = false
+	letsEncryptProd    = "acme-v02.api.letsencrypt.org"
+	letsEncryptStaging = "acme-staging-v02.api.letsencrypt.org"
 )
 
-type myRule struct {
-	addrs []net.IP
-	fqdns []string
+type allowLetsEncrypt struct{}
```

Comment:
refactoring the old rule to be a little more specific.  While the old setup was more general, we didn't really need it and made making some sort of uniform behavior between swapping multiple rule sets more complex than needed.